modules/viewHierarchyMixin.js

```diff
@@ -106,49 +111,201 @@
+     */
+    deactivateCallback: function() {
+      // do nothing
+    },
+
+    /**
+     * Method to be invoked when activate is called. Use this method to turn on any
+     * custom timers, listenTo's or on's that should be activatable.
+     * @method deactivateCallback
+     */
+    activateCallback: function() {
+      // do nothing
+    },
+
+    /**
      * Default child view cleanup method that may be overriden.
      * @method cleanupChildViews
      */
     cleanupChildViews: function() {
-      // do nothing
+      _.each(this._childViews, function(view) {
+        view.dispose();
+      }, this);
+    },
+
+    /**
+     * Deactivates all child views
+     * Default method may be overriden.
+     * @method deactivateChildViews
+     */
+    deactivateChildViews: function() {
+      _.each(this._childViews, function(view) {
+        view.deactivate();
+      }, this);
+    },
+
+    /**
+     * Activates all child views
+     * Default method may be overriden.
+     * @method deactivateChildViews
+     */
+    activateChildViews: function() {
+      _.each(this._childViews, function(view) {
+        view.activate();
+      }, this);
+    },
+
+    /**
+     * Binds the view as a child view - any recursive calls like activate, deactivate, or dispose will
+     * be done to the child view as well.
+     * @param view {View} the child view
+     * @return {View} the child view
+     * @method registerChildView
+     */
+    registerChildView: function(view) {
+      this._childViews[view.cid] = view;
+      return view;
+    },
+
+    /**
+     * Unbinds the child view - no recursive calls will be made to this child view
+     * @param view {View} the child view
+     * @return {View} the child view
+     * @method unregisterChildView
+     */
+    unregisterChildView: function(view) {
+      delete this._childViews[view.cid];
+      return view;
     },
 
     /**
-     * Injects a child view and triggers a re-render on that view
+     * Attaches a child view by finding the element with the attribute inject=<injectionSite>
+     * Invokes attachChildView as the bulk of the functionality
      * @method injectView
      * @param injectionSite {String} The name of the injection site in the layout template
      * @param view          {View}   The instantiated view object to inject
      */
     injectView: function(injectionSite, view) {
       var injectionPoint = this.$el.find('[inject=' + injectionSite + ']');
-
       if (view && injectionPoint) {
-        injectionPoint.html(view.$el);
-        view.render();
+        this.attachChildView(injectionPoint, view);
+      }
+    },
+
+    /**
+     * Registers the child view if not already done so, then calls view.attach with the element argument
+     * @param $el {jQuery element} the element to attach to.
+     * @param view {View} the child view
+     * @method attachChildView
+     */
+    attachChildView: function($el, view) {
+      if (!this._childViews[view.cid]) {
```

Comment:
is there any advantage of doing this if check before registering, could you just call register every time?